cartographer/mapping/internal/3d/scan_matching/intensity_cost_function_3d.cc

```diff
@@ -0,0 +1,25 @@
+#include "cartographer/mapping/internal/3d/scan_matching/intensity_cost_function_3d.h"
+
+namespace cartographer {
+namespace mapping {
+namespace scan_matching {
+
+// This method is defined here instead of the header file as it was observed
+// that defining it in the header file has a negative impact on the runtime
+// performance.
```

Comment:
Interesting, do you know if this is something that only applies to this function or if it also affects the other cost functions?

Reply:
No idea. But a lot of attention was paid to make sure that this does not have a measurable effect on running time performance when intensities are not used.